api/controllers/game/ready.js

```diff
@@ -31,7 +31,9 @@ module.exports = async function (req, res) {
       sails.sockets.blast('gameStarted', { gameId: game.id });
       gameUpdates.status = gameService.GameStatus.STARTED;
       // Deal cards (also emits socket event)
-      await gameService.dealCards(game, gameUpdates);
+      await Game.updateOne({ id: game.id }).set(gameUpdates);
+
+      await gameService.dealCards({...game, ...gameUpdates}, {});
```

Comment:
Because dealing the cards and  other logic for starting the game require updating the Game row, the legacy utility for dealing the card's inputs a dictionary of changes to apply during the dealCards() function, so that the Game can be updated once instead of twice. 

Since the GameState API version of `dealCards()` just creates a new (initial) GameState, it doesn't require updating the Game, so it makes sense to update the Game's status and the p0Ready/p1Ready columns *before* calling `dealCards()`.

This has the awkward side effect, at least in the way I'm doing it now, of making two game updates in the legacy (actual production) environment, since now we're updating the game here and then again. It feels awkward, and I'm not sure what's the best way to resolve it. 

It might be cleaner if we managed the update in gsmeService.dealCards() in the section where we bifurcate on the env var

Reply:
I explored reworking this to save game updates in the same helper, but either way we'll need to rework rematch to make this work, so let's set that aside until we get to rematch